packages/core-flows/src/fulfillment/steps/delete-fulfillment-sets.ts

```diff
@@ -0,0 +1,28 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IFulfillmentModuleService } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const deleteFulfillmentSetsStepId = "delete-fulfillment-sets"
+export const deleteFulfillmentSetsStep = createStep(
+  deleteFulfillmentSetsStepId,
+  async (ids: string[], { container }) => {
+    const service = container.resolve<IFulfillmentModuleService>(
+      ModuleRegistrationName.FULFILLMENT
+    )
+
+    await service.softDelete(ids)
+
+    return new StepResponse(void 0, ids)
```

Comment:
```suggestion
    const deleted = await service.softDelete(ids)

    return new StepResponse({
      [Modules.FULFILLMENT]: deleted,
    }, ids)
```

Reply:
This return value seems very specific to something we want to do subsequently, that is remove the links. I would argue we should return the IDs and transform the data to the link-shape separately..